### ISSUE TITLE
Allow settings blocks

### ISSUE THREAD

**Issue by kroppt** [microsoft/vscode#79524]
TL;DR allow duplicate settings keys with blocks

I am currently creating an extension which is going to force settings for dependent extensions in order to provide a certain default environment. I think a good addition to the settings file would be the ability to create separate blocks that don't overwrite other entries.

I think it would be useful to allow for users to add their own settings without immediately being overwritten.

Example `settings.json`:
```
{
    {
        "editor.renderWhitespace": "all",
    },
    "editor.renderWhitespace": "boundary",
}
```

The lower-depth setting takes precedence.

This change would probably affect the way the settings API works for extensions authors, as there would have to be a way to access settings blocks.

The naming of these blocks would be useful, too, and would be similar to language-specific setting notation.

Example `settings.json`:
```
{
    "my.extension" {
        "editor.renderWhitespace": "all",
    },
}
```

**Comment by RMacfarlane** [comment]
We currently allow scoping settings to for specific languages: https://code.visualstudio.com/docs/getstarted/settings#_language-specific-editor-settings

Besides that, settings can be scoped to a workspace or folder, or overwritten from an extension.

I think introducing this will cause confusion both for users and extension authors - instead of accessing a setting by a key, you now have to have a concept of the depth at which you want to write it. And as a user, you would need to understand yet another level of precedence. I don't think we want to duplicate settings in an arbitrary way, so I'm marking this as out of scope. If you still think this is something that should be done, please explain your use case a bit more.
